clarity/src/proptesting/types.rs

```diff
@@ -0,0 +1,78 @@
+
+use super::*;
+use crate::types::{StacksHashMap, StacksHashSet};
+use crate::vm::callables::{DefineType, DefinedFunction, FunctionIdentifier};
+use crate::vm::contracts::Contract;
+use crate::vm::representations::{
+    Span, SymbolicExpression, SymbolicExpressionType, TraitDefinition,
+};
+use crate::vm::types::{
+    ASCIIData, BuffData, CharType, ListData, ListTypeData, OptionalData, PrincipalData,
+    QualifiedContractIdentifier, ResponseData, SequenceData, SequenceSubtype,
+    StandardPrincipalData, StringSubtype, StringUTF8Length, TraitIdentifier, TupleData,
+    TupleTypeSignature, TypeSignature, UTF8Data, Value, MAX_VALUE_SIZE,
+};
+use crate::vm::{ClarityName, ClarityVersion, ContractContext, ContractName};
+
+pub fn standard_principal_data() -> impl Strategy<Value = StandardPrincipalData> {
+    (0u8..32, prop::collection::vec(any::<u8>(), 20))
+        .prop_map(|(v, hash)| StandardPrincipalData(v, hash.try_into().unwrap()))
+}
+
+pub fn qualified_contract_identifier() -> impl Strategy<Value = QualifiedContractIdentifier> {
+    (standard_principal_data(), contract_name())
+        .prop_map(|(issuer, name)| QualifiedContractIdentifier { issuer, name })
+}
+
+pub fn trait_identifier() -> impl Strategy<Value = TraitIdentifier> {
+    (clarity_name(), qualified_contract_identifier()).prop_map(|(name, contract_identifier)| {
+        TraitIdentifier {
+            name,
+            contract_identifier,
+        }
+    })
+}
+
+pub fn type_signature() -> impl Strategy<Value = TypeSignature> {
+    let leaf = prop_oneof![
+        Just(TypeSignature::IntType),
+        Just(TypeSignature::UIntType),
+        Just(TypeSignature::BoolType),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::BufferType(
+            s.try_into().unwrap()
+        ))),
+        (0u32..128).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::StringType(
+            StringSubtype::ASCII(s.try_into().unwrap())
+        ))),
+        Just(TypeSignature::PrincipalType),
+        (0u32..32).prop_map(|s| TypeSignature::SequenceType(SequenceSubtype::StringType(
+            StringSubtype::UTF8(s.try_into().unwrap())
+        )))
+    ];
+
+    leaf.prop_recursive(3, 32, 5, |inner| prop_oneof![
+        // optional type: 10% NoType + 90% any other type
+        prop_oneof![
+            1 => Just(TypeSignature::NoType),
+            9 => inner.clone(),
+        ]
+        .prop_map(|t| TypeSignature::new_option(t).unwrap()),
+        // response type: 20% (NoType, any) + 20% (any, NoType) + 60% (any, any)
+        prop_oneof![
+            1 => inner.clone().prop_map(|ok_ty| TypeSignature::new_response(ok_ty, TypeSignature::NoType).unwrap()),
+            1 => inner.clone().prop_map(|err_ty| TypeSignature::new_response(TypeSignature::NoType, err_ty).unwrap()),
+            3 => (inner.clone(), inner.clone()).prop_map(|(ok_ty, err_ty)| TypeSignature::new_response(ok_ty, err_ty).unwrap()),
+        ],
```

Comment:
All those probabilities to create `NoType` were designed to check on the limitations of the typechecker in clarity-wasm. I'm not sure this is needed for your use case.

Reply:
Tbh I will probably need to add different strategies for different use-cases, like clarity1 vs clarity2, etc. since i.e. trait handling is different between the versions, and other nuances.

Reply:
OK to resolve this with the same reasons as the other comment? Not sure how this functionality will evolve as more property tests are added to `clarity`.

Reply:
Sure. Personally I don't mind. I was just not sure how useful it was for you, and if it was worth it to handle the `NoType` in your code.